pkg/api/internalclient/billing.go

```diff
@@ -0,0 +1,46 @@
+package internalclient
+
+import (
+	"net/http"
+
+	"github.com/shellhub-io/shellhub/pkg/models"
+)
+
+// billingAPI defines methods for interacting with billing-related functionality.
+type billingAPI interface {
+	// BillingReport sends a billing report for the specified tenant and action.
+	// It returns the HTTP status code of the response and an error, if any.
+	BillingReport(tenant string, action string) (int, error)
+
+	// BillingEvaluate evaluates the billing status for the specified tenant.
+	// It returns the billing evaluation result, the HTTP status code of the response, and an error, if any.
+	BillingEvaluate(tenantID string) (*models.BillingEvaluation, int, error)
+}
+
+func (c *client) BillingReport(tenant string, action string) (int, error) {
+	res, err := c.http.
+		R().
+		SetHeader("X-Tenant-ID", tenant).
+		SetQueryParam("action", action).
+		Post("http://billing-api:8080/internal/billing/report")
```

Comment:
Would you mind confirming if the request built won't be something like `http://api:8080/http://billing-api:8080/internal/billing/report` due to usage of the same client what has `http://api:8080` as URL?

Reply:
According to the [official documentation](https://pkg.go.dev/github.com/go-resty/resty/v2#Client.SetBaseURL), SetBaseURL is only intended for relative URLs.

Reply:
Exactly, but you are using the same client as you have set the URL, so It could end up with something like what I have mentioned.

Reply:
I understand, but even when using the same client, the BaseURL should only work with relative URLs:
> It will be used with request raised from this client **with** relative URL

Reply:
ust to confirm, I've conducted some tests, and yes, it does not use the base URL when the request is something like `http://billing-api:8080/internal/billing/report`~.

Reply:
I have checked and the problem I have mention seems does not happen. Perfect!